fontbe/src/orchestration.rs

```diff
@@ -219,8 +219,67 @@ impl From<&VariationRegion> for TupleBuilder {
     }
 }
 
+const SHORT_LOCA: u8 = 0;
+const LONG_LOCA: u8 = 1;
+
 pub type BeWork = dyn Work<Context, Error> + Send;
-type GlyfLoca = (Vec<u8>, Vec<u32>);
+pub struct GlyfLoca {
+    pub glyf: Vec<u8>,
+    pub loca: Vec<u32>,
+    pub format: u8,
+}
+
+impl GlyfLoca {
+    pub fn new(glyf: Vec<u8>, loca: Vec<u32>) -> Self {
+        // https://github.com/fonttools/fonttools/blob/1c283756a5e39d69459eea80ed12792adc4922dd/Lib/fontTools/ttLib/tables/_l_o_c_a.py#L37
+        let mut format = SHORT_LOCA;
+        if let Some(max_offset) = loca.last() {
+            if *max_offset >= 0x20000 || loca.iter().any(|offset| offset % 2 != 0) {
+                format = LONG_LOCA;
+            }
+        }
```

Comment:
I really go out of my way to avoid this pattern ("declare a mut default and set it in a branch") where possible, because I think in general it is harder to follow.

In this case I think it probably makes sense to just have a standalone `compute_loca_format` fn?

Reply:
Fixed per IM code golf